planner/routers/base_router.py

```diff
@@ -1,8 +1,7 @@
-import abc
+from typing import NamedTuple
 
 
-class BaseRouter(abc.ABC):
-
-    @abc.abstractmethod
-    def make_route(self, origin, destination, *args, **kwargs):
-        raise NotImplementedError
+class RouteResult(NamedTuple):
```

Comment:
Why subclass NamedTuple? Not cannonical, see https://docs.python.org/3/library/collections.html#collections.namedtuple.

Also just an object is okay

Reply:
This is pretty tricky...https://stackoverflow.com/questions/5906831/serializing-a-python-namedtuple-to-json

Reply:
https://docs.python.org/3/library/typing.html#typing.NamedTuple